tracer/src/Datadog.Trace/Agent/Transports/HttpClientRequest.cs

```diff
@@ -22,18 +22,26 @@ internal class HttpClientRequest : IApiRequest
 
         private readonly HttpClient _client;
         private readonly HttpRequestMessage _request;
+        private readonly Uri _uri;
 
         public HttpClientRequest(HttpClient client, Uri endpoint)
         {
             _client = client;
             _request = new HttpRequestMessage(HttpMethod.Post, endpoint);
+            _uri = endpoint;
         }
 
         public void AddHeader(string name, string value)
         {
             _request.Headers.Add(name, value);
         }
 
+        public async Task<IApiResponse> GetAsync()
+        {
+            _request.Method = HttpMethod.Get;
```

Comment:
After calling this method, using `PostAsync()` will use `GET` instead of `POST`. We should probably just `_request.Method` explicitely in each method instead of relying of the verb set in the ctor.

Reply:
https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=105050&view=ms.vss-test-web.build-test-results-tab&runId=7420872&resultId=100039&paneView=debug

Tests started to fail after this change: 29c08d5d3a11659ac9e708b37ca61ac8c77b20aa.
Apparently, the setter method of `Method` in .NET Core (which is `HttpRequestMessage.set_Method`) is creating a new `traceId`.

Reply:
LGTM as a temporary fix. Thanks!